src/preferences/PreferencesBase.js

```diff
@@ -189,7 +203,13 @@ define(function (require, exports, module) {
                         try {
                             result.resolve(JSON.parse(text));
                         } catch (e) {
-                            result.reject(new ParsingError("Invalid JSON settings at " + path + "(" + e.toString() + ")"));
+                            if (isStateJson && createIfNew) {
```

Comment:
Where do we create this file when `createIfNew` is true. I am unclear what is the meaning of this flag createIfNew. From its name it appears that we will create a new file after checking if one does not already exist. I can prefs file is getting updated in `save` but `createIfNew` is not being checked there